From 1771755a901e2054d57af7603012cf12b5f1bb76 Mon Sep 17 00:00:00 2001
From: Eike Ziller <eike.ziller@theqtcompany.com>
Date: Mon, 22 Jun 2015 17:39:00 +0200
Subject: [PATCH] OS X: Do not use too recent API for spotlight locator filter

NSMetadataQueryUpdateAddedItemsKey is only available on 10.9 or later
and is available through Foundation only in 10.10.
Use the CoreServices constant instead, which is available since 10.4.

Task-number: QTCREATORBUG-14525
Change-Id: Icddedb77c77efca82ca00090c8a5d53db3284c96
---
 src/plugins/coreplugin/locator/spotlightlocatorfilter.mm | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/plugins/coreplugin/locator/spotlightlocatorfilter.mm b/src/plugins/coreplugin/locator/spotlightlocatorfilter.mm
index 4b63274..cb29e2c 100644
--- src/plugins/coreplugin/locator/spotlightlocatorfilter.mm
+++ src/plugins/coreplugin/locator/spotlightlocatorfilter.mm
@@ -45,6 +45,8 @@
 #include <Foundation/NSOperation.h>
 #include <Foundation/NSPredicate.h>
 
+#include <CoreServices/CoreServices.h>
+
 namespace Core {
 namespace Internal {
 
@@ -96,7 +98,7 @@ SpotlightIterator::SpotlightIterator(const QString &expression)
                     usingBlock:^(NSNotification *note) {
                         [m_query disableUpdates];
                         QMutexLocker lock(&m_mutex); Q_UNUSED(lock)
-                        [m_queue addObjectsFromArray:[note.userInfo objectForKey:NSMetadataQueryUpdateAddedItemsKey]];
+                        [m_queue addObjectsFromArray:[note.userInfo objectForKey:(NSString *)kMDQueryUpdateAddedItems]];
                         [m_query enableUpdates];
                         m_waitForItems.wakeAll();
                     }] retain];
-- 
2.4.3

